draft-ietf-jsonpath-base.md

```diff
@@ -577,207 +590,227 @@ of node.
 
+Note that the `dot-selector` follows the philosophy of JSON strings and is
+allowed to contain bit sequences that cannot encode Unicode characters (a
+single unpaired UTF-16 surrogate, for example).
+The behaviour of an implementation is undefined for member names which do
+not encode Unicode characters.
 
-#### Child {#unionchild}
+### Dot Wild Card Selector
 
-##### Syntax
+#### Syntax
 {: numbered="false" toc="exclude"}
 
-A child is a quoted string.
+The dot wild card selector has the shape `'.*'`.
 
 ~~~~ abnf
-union-element = child ; see below for more alternatives
-child = %x22 *double-quoted %x22 / ; "string"
-        "'" *single-quoted "'"   ; 'string'
+dot-wild-selector    = "." "*"            ;  dot followed by asterisk
+~~~~
+
+#### Semantics
+{: numbered="false" toc="exclude"}
+
+A dot selector combined with a single asterisk is a wild card. It selects all member values of an object as well as all elements of an array. Applying the `dot-wild-selector` to primitive JSON values, as number, string, or literal, does nothing.
 
-double-quoted = dq-unescaped /
-          escape (
-              %x22 /         ; "    quotation mark  U+0022
-              "/" /          ; /    solidus         U+002F
-              "\" /          ; \    reverse solidus U+005C
-              "b" /          ; b    backspace       U+0008
-              "f" /          ; f    form feed       U+000C
-              "n" /          ; n    line feed       U+000A
-              "r" /          ; r    carriage return U+000D
-              "t" /          ; t    tab             U+0009
-              "u" 4HEXDIG )  ; uXXXX                U+XXXX
 
+### Index Selector
 
-dq-unescaped = %x20-21 / %x23-5B / %x5D-10FFFF
+#### Syntax
+{: numbered="false" toc="exclude"}
 
-single-quoted = sq-unescaped /
-          escape (
-              "'" /          ; '    apostrophe      U+0027
-              "/" /          ; /    solidus         U+002F
-              "\" /          ; \    reverse solidus U+005C
-              "b" /          ; b    backspace       U+0008
-              "f" /          ; f    form feed       U+000C
-              "n" /          ; n    line feed       U+000A
-              "r" /          ; r    carriage return U+000D
-              "t" /          ; t    tab             U+0009
-              "u" 4HEXDIG )  ; uXXXX                U+XXXX
+An index selector `'[<index>]'` addresses a single object member value or a single array element value.
 
-sq-unescaped = %x20-26 / %x28-5B / %x5D-10FFFF
+~~~~ abnf
+index-selector      = "[" (quoted-member-name / element-index) "]"
+~~~~
 
-escape = "\"                 ; \
+Applying the `index-selector` to an object value, a `quoted-member-name` string is required. JSONPath allows it to be enclosed in _single_ or _double_ quotes.
 
-HEXDIG =  DIGIT / "A" / "B" / "C" / "D" / "E" / "F"
-                              ; case insensitive hex digit
+~~~~ abnf
+quoted-member-name  =  %x22 *double-quoted %x22 /       ; "string"
```

Comment:
I think we should have a separate set of definitions for literals.  This is particular important because JSONPath literals have a more permissive syntax than that of JSON.  (Should they?)
Literals should never be confused with the data themselves (i.e., we never should say "string" when we mean "string literal").

Reply:
> Literals should never be confused with the data themselves (i.e., we never should say "string" when we mean "string literal").

What?  When was this discussed?  Do we now have to say "boolean literal" or "numeric literal?"  Where does JSON Path ever use strings that AREN'T a string literal?